File: src/main/kotlin/com/lambda/client/event/events/client/StepEvent.java
package me.han.muffin.client.event.events.client;

import me.han.muffin.client.event.EventCancellable;

public class StepEvent extends EventCancellable {
    private final EventStage stage;
    private float height;

    public StepEvent(EventStage stage ,float height) {
        this.stage = stage;
        this.height = height;
    }


    public float getHeight() {
        return height;
    }

    public void setHeight(float height) {
        this.height = height;
    }

    @Override
    public EventStage getStage() {
        return stage;
    }
}

File: src/main/kotlin/com/lambda/client/module/modules/events/StepEvent.kt
package com.lambda.client.module.modules.events

import com.lambda.client.module.modules.EventStage
import net.minecraft.entity.Entity
import net.minecraftforge.fml.common.eventhandler.Cancelable

@Cancelable
class StepEvent(stage: Int, val entity: Entity) : EventStage(stage) {
    var height: Float

    init {
        height = entity.stepHeight
    }
}